docs/db-core-interface.md

```diff
@@ -1,32 +1,68 @@
+# ODC Enhancement: Replace the ODC Index and API
+
+**Date:** 2019-06-12
+
+**Author:** Kirill Kouzoubov  (@Kirill888), Damien Ayers (@omad), Alex Leith (@alexgleith)
+
+**Version:** datacube-core >=  1.7
+
 # Introduction
 
-**This is an evolving document**
+The Database API is used in the ODC Core for all access to the ODC Index. The Index
+is intended to be very flexible in the metadata formats it supports, while also
+maintaining high performance searching across multiple dimensions.
+
+Mostly, it has been successful in these goals. Unfortunately, while trying to
+implement improvements like fast extents queries (link), native resolution data
+loading (insert link) and other smaller changes, we have run up against several
+assumptions and design flaws which are limiting future ODC development.
+
+This proposal is for a replacement of the ODC Index and Database API and
+a change in the internal model of *Datasets*.
+
+## Summary
+
+Instead of storing heterogeneous dataset structures in the database with a lot of
+flexibility in what is indexed (in a database index sense), a new database
+structure with required fields stored as database columns, and only ancilliary
```

Comment:
We are not trying to mandate DB structure, rather we are aiming to define an internal API between core and DB that would allow for more efficient DB structures, particularly for spatial data.

Reply:
Ok, I seem to have over simplified. I'll need to talk to you and Damien and get on the same page.